NBitcoin/Protocol/NodesCollection.cs

```diff
@@ -64,6 +64,29 @@ public void PushMessage(IncomingMessage message)
 
 			#endregion
 		}
+
+		/// <summary>
+		/// Provides a comparer to specify how nodes are compared for equality
+		/// </summary>
+		public class NodeComparer:IEqualityComparer<Node>
+		{
+			public bool Equals(Node nodeA, Node nodeB)
+			{
+				if (nodeA == null || nodeB == null)
+					return nodeA == null && nodeB == null;
+
+				return nodeA.RemoteSocketAddress.MapToIPv6().ToString() == nodeB.RemoteSocketAddress.MapToIPv6().ToString() && nodeA.RemoteSocketPort == nodeB.RemoteSocketPort;
```

Comment:
Do you really need ToString IPAddres I think it handles equality internally.

Reply:
Would it not be an idea to give the Node class some kind of unchanging ID property that can be used here and elsewhere, even if it just maps to

public  Id => nodeA.RemoteSocketAddress.MapToIPv6().ToString() + nodeA.RemoteSocketPort  ?

here could be simplified to:  return nodeA.Id == nodeB.Id

Reply:
The node has already an id (the nonce) your suggestion is nice but apart for this usage will we need it else where?

I am going to commit this as is, if you want to submit a PR that implements your suggestion it will be welcome.